source/tyr/tyr/binarisation.py

```diff
@@ -624,6 +624,46 @@ def bano2mimir(self, autocomplete_instance, filename, job_id, dataset_uid):
         raise
 
 
+@celery.task(bind=True)
+def openaddresses2mimir(self, autocomplete_instance, filename, job_id, dataset_uid):
+    """ launch openaddresses2mimir """
+    autocomplete_instance = models.db.session.merge(autocomplete_instance)  # reatache the object
+    logger = get_autocomplete_instance_logger(autocomplete_instance, task_id=job_id)
+    job = models.Job.query.get(job_id)
+    cnx_string = current_app.config['MIMIR_URL']
+    working_directory = unzip_if_needed(filename)
+
+    if autocomplete_instance.address != 'OA':
+        logger.warn(
+            'no open addresses data will be loaded for instance {} because the address are read from {}'.format(
+                autocomplete_instance.name, autocomplete_instance.address
+            )
+        )
+        return
+
+    try:
+        res = launch_exec(
+            "openaddresses2mimir",
+            [
+                '-i',
+                working_directory,
+                '--connection-string',
+                cnx_string,
+                '--dataset',
+                autocomplete_instance.name,
+            ],
+            logger,
+        )
+        if res != 0:
+            # @TODO: exception
```

Comment:
Why it is a TODO ? Because you already raise a valueError.

Reply:
ahah I used the good ol' copy paste method for this, and all the other tasks have this TODO :roll_eyes: :see_no_evil: 

I think this TODO is to better type the raised exception and not use a generic exception